src/core/ext/lb_policy/grpclb/grpclb.c

```diff
@@ -486,6 +629,11 @@ static void glb_destroy(grpc_exec_ctx *exec_ctx, grpc_lb_policy *pol) {
     grpc_grpclb_destroy_serverlist(glb_policy->serverlist);
   }
   gpr_mu_destroy(&glb_policy->mu);
+
+  for (size_t i = 0; i < glb_policy->num_ok_serverlist_addresses; ++i) {
+    user_data_destroy(glb_policy->lb_addresses[i].user_data);
+  }
+  gpr_free(glb_policy->lb_addresses);
```

Comment:
It happened in `create_rr`. I've moved everything into the new `lb_addrs_destroy`.